### PR TITLE
feat(rn) update React Native to version 0.66

### PR DESCRIPTION
**Pull request by saghul**

### CODE REVIEW

File: android/sdk/src/debug/AndroidManifest.xml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="utf-8"?>
+<manifest xmlns:android="http://schemas.android.com/apk/res/android">
+    <uses-permission android:name="android.permission.SYSTEM_ALERT_WINDOW"/>
+
+    <application android:usesCleartextTraffic="true">
```
[the remark this answers is not in the document]
Comment (saghul) [reply android/sdk/src/debug/AndroidManifest.xml:5]: In debug, yes. The metro bundled runs on HTTP. Come tho think of it, we have this already set in the network XML, I'll check if we can simplify.

**Comment by saghul** [comment]
Jenkins please test this please.

**Review by Calinteodor — APPROVED** [review]

**Comment by Calinteodor** [comment]
Everything seems ok. iOS simulator doesn't act funky. 
One thing, when I build on Android studio, I get this error.
image

**Comment by saghul** [comment]
> One thing, when I build on Android studio, I get this error.

Thanks for testing Calin! You need to configure Android Studio to use Java 11. Mine was configured to use 8 by default which caused that error.

**Comment by saghul** [comment]
I'll look into the test failures tomorrow, maybe the React update triggered something.

**Comment by Calinteodor** [comment]
> > One thing, when I build on Android studio, I get this error.
> 
> Thanks for testing Calin! You need to configure Android Studio to use Java 11. Mine was configured to use 8 by default which caused that error.

Configured and it works. Thanks for the hint!

**Comment by diegolmello** [comment]
@saghul any change of enabling Hermes?

File: android/sdk/src/debug/AndroidManifest.xml (hunk 1 above)
[the remark this answers is not in the document]
Comment (saghul) [reply android/sdk/src/debug/AndroidManifest.xml:5]: I double checked, we are ok.

**Comment by diegolmello** [comment]
Thanks, @saghul! ❤️

**Review by Calinteodor — APPROVED** [review]

**Pull request merged**